sdk/android/iproov_firebase/src/main/java/com/iproov/firebase/IProovFirebase.kt

```diff
@@ -26,108 +26,114 @@ enum class ClaimType(val value: String) {
     VERIFY("verify"),
 }
 
-suspend fun FirebaseAuth.signInWithIProov(
-    applicationContext: Context,
-    userId: String,
-    iProovEvents: MutableStateFlow<IProov.IProovSessionState?>? = null,
-    assuranceType: AssuranceType = AssuranceType.GENUINE_PRESENCE,
-    iProovOptions: IProov.Options = IProov.Options(),
-    extensionId: String = defaultExtensionId,
-) {
-    return doIProov(
-        applicationContext,
-        userId,
-        iProovEvents,
-        assuranceType,
-        ClaimType.VERIFY,
-        iProovOptions,
-        extensionId,
-    )
-}
+fun FirebaseAuth.iProov(region: String? = null, extensionId: String? = null)= IProovFirebaseAuth(this, extensionId = extensionId ?: defaultExtensionId, region = region)
 
-suspend fun FirebaseAuth.createIProovUser(
-    applicationContext: Context,
-    userId: String,
-    iProovEvents: MutableStateFlow<IProov.IProovSessionState?>? = null,
-    assuranceType: AssuranceType = AssuranceType.GENUINE_PRESENCE,
-    iproovOptions: IProov.Options = IProov.Options(),
-    extensionId: String = defaultExtensionId,
-) {
-    return doIProov(
-        applicationContext,
-        userId,
-        iProovEvents,
-        assuranceType,
-        ClaimType.ENROL,
-        iproovOptions,
-        extensionId,
-    )
-}
+class IProovFirebaseAuth(private val functions: FirebaseFunctions, private val auth: FirebaseAuth, private val extensionId: String){
 
-private suspend fun doIProov(
-    applicationContext: Context,
-    userId: String,
-    iProovEvents: MutableStateFlow<IProov.IProovSessionState?>?,
-    assuranceType: AssuranceType,
-    claimType: ClaimType,
-    iproovOptions: IProov.Options,
-    extensionId: String,
-) {
-    val response =
-        FirebaseFunctions.getInstance()
-            .getHttpsCallable("ext-${extensionId}-getToken")
-            .call(
+    constructor(auth: FirebaseAuth, extensionId: String? = null, region: String? = null) :
+            this(FirebaseFunctions.getInstance(auth.app, region ?: defaultRegion ), auth, extensionId = extensionId ?: defaultExtensionId,)
```

Comment:
formatting:

```suggestion
            this(FirebaseFunctions.getInstance(auth.app, region ?: defaultRegion), auth, extensionId = extensionId ?: defaultExtensionId)
```

(consider whether you want to leave the trailing comma, but [while kotlin style guide recommends it at declaration, it's considered discretionary at the call-site](https://kotlinlang.org/docs/coding-conventions.html#trailing-commas))